tests/unit/scripting.tcl

```diff
@@ -430,6 +430,33 @@ start_server {tags {"scripting"}} {
         set res
     } {102}
 
+    test {EVAL timeout from AOF} {
+        # generate a long running script that is propagated to the AOF as script
+        # make sure that the script times out during loading
+        r config set appendonly no
+        r config set aof-use-rdb-preamble no
+        r config set lua-replicate-commands no
+        r flushall
+        r config set appendonly yes
+        wait_for_condition 50 100 {
+            [s aof_rewrite_in_progress] == 0
+        } else {
+            fail "AOF rewrite can't complete after CONFIG SET appendonly yes."
+        }
+        r config set lua-time-limit 1
+        set rd [redis_deferring_client]
+        $rd eval {redis.call('set',KEYS[1],'y'); for i=1,1000000 do redis.call('ping') end return 'ok'} 1 x
```

Comment:
I'm a bit concerned about the flakiness of this, did you consider other options?
(I don't have a better suggestion that doesn't involve creating new synthetic delay commands that are script-accessible)

Reply:
i can't think of any other option.
scripts are by definition immune from being affected by external factors.
DEBUG can't be used in scripts.
and SCRIPT KILL can't be used since this one has to have a write command in it.
the only thing i can think of is find some command that has a deterministic timing.
but obviously blocked commands with timeout are not allowed either.

maybe we can settle on the fact i was able to reproduce it when i fixed it, and we can drop the test now.
or maybe keep the test, but remove the assertion that makes sure it was able to trigger the BUSY and LOADING state. (so that it won't be flaky)

Reply:
i guess i was missing the `rd flush` and `after 100` to make it work.
added some timing to see what's going on, and added more time so that on my machine it took 900ms for each eval.
on actions it took:
```
script took 20125 milliseconds
loading took 19310 milliseconds
```
i guess we can cut it by half?

Reply:
with the change it takes some 400ms on my machine, and 10s on github actions
```
script took 10532 milliseconds
loading took 11399 milliseconds
```